src/simulators/matrix_product_state/matrix_product_state_internal.cpp

```diff
@@ -573,21 +573,40 @@ void MPS::apply_swap_internal(uint_t index_A, uint_t index_B, bool swap_gate) {
 void MPS::apply_2_qubit_gate(uint_t index_A, uint_t index_B, Gates gate_type, const cmatrix_t &mat, bool is_diagonal)
 {
   // We first move the two qubits to be in consecutive positions
-  // If index_B > index_A, we move the qubit at index_B to index_A+1
-  // If index_B < index_A, we move the qubit at index_B to index_A-1, and then
-  // swap between the qubits
+  // For performance reasons, we move the qubit with higher bond dimension
+  // toward the qubit with lower bond dimension
+  // Case i:
+  //     If index_B > index_A, we either move the qubit at index_B to index_A+1
+  //                                 or  move the qubit at index_A to index_B-1
+  // Case ii:
+  //     If index_B < index_A, we either move the qubit at index_B to index_A-1, 
+  //                                  or move the qubit at index_A to index_B+1
```

Comment:
In the documentation and in the code, you can reduce the two cases into a single case by calling a common function

Reply:
I took your suggestion. Indeed, it looks better now. Let me know what you think.